### PR TITLE
Updates to the release notes scripts and documentation

### PR DESCRIPTION
**Pull request by drisspg**
# Summary
This PR made some significant changes to the scripts around Release Scripts. At a high level:
- Turned the quips into docs and updated links
- Update the common.categorizes list in the hopes to make this the source of truth for releases- This is hard since the release_notes labels can be changed at will. An alternative would be to poll from github api. However, I think that is overkill. The notebook does a set compare and will show you knew categories. I think we want this to be manual so that the release note engineer will decided how to categorize.
- Create cateogry group from speaking with folks on distributed and AO that told me these different release categories can be merged.
- I am the newest person to Core and don't use ghstack soo made token getting a lil more generic.
- Added a classifier.py file. This file will train a commit categorizer for you, hopefully with decent accuracy. I was able to achieve 75% accuracy. I drop the highest frequency class - "skip" since this creates a more useful cateogrizer.
- I updated the categorize.py script so that the prompt will be what the classifier thinks, gated by a flag.
- Added a readme that will hopefully help future release notes engineers.

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: scripts/release_notes/classifier.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+        title_embed = title_features[:, 0, :]
+        title_embed = self.dropout(title_embed)
+        title_embed = self.dense_title(title_embed)
+        title_embed = self.activation_fn(title_embed)
+        title_embed = self.dropout(title_embed)
+        title_embed = self.out_proj_title(title_embed)
+
+        # Encode file input
+        # IF someone wants to make this better:
+        # I think a better feature would be splitting file strings to
```
Comment (drisspg) [inline scripts/release_notes/classifier.py:99]: I did this, I can remove this, added a 10% val set accuracy win

File: scripts/release_notes/common.py [hunk 2]
```diff
@@ -33,20 +85,16 @@
...
-    'dataloader_frontend',
-    'sparse_frontend'
-]
+    # 2.0 release
+    'mps',
+    'intel',
+    'functorch',
+    'gnn',
+    'distributions',
+    'windows',
```
Comment (drisspg) [inline scripts/release_notes/common.py:94]: this should really be signals if we want it.

File: scripts/release_notes/classifier.py [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+    categories: List[str]
+    input_dim: int = 768
+    inner_dim: int = 128
+    dropout: float = 0.1
+    activation = nn.ReLU
+    embedding_dim: int = 8
+    file_embedding_dim: int = 32
+
+
+class CommitClassifier(nn.Module):
```
Comment (albanD) [inline scripts/release_notes/classifier.py:64]: Should we replace this with the openai API for chatgpt? :D

**Review by albanD — COMMENTED** [review]
Didn't go through all of this in detail but sounds good. I'm sure you'll guide the next person having to use this!

**Review by albanD — APPROVED** [review]
